Documentation/Diagnostics/PH2101.md

```diff
@@ -0,0 +1,47 @@
+# PH2101: Detect Null Dereference after "as"
+
+| Property | Value  |
+|--|--|
+| Package | [Philips.CodeAnalysis.MaintainabilityAnalyzers](https://www.nuget.org/packages/Philips.CodeAnalysis.MaintainabilityAnalyzers) |
+| Diagnostic ID | PH2101 |
+| Category  | [RuntimeFailure](../RuntimeFailure.md) |
+| Analyzer | [DereferenceNullAnalyzer](https://github.com/philips-software/roslyn-analyzers/blob/master/Philips.CodeAnalysis.MaintainabilityAnalyzers/RuntimeFailure/DereferenceNullAnalyzer.cs)
+| CodeFix  | No |
+| Severity | Error |
+| Enabled By Default | Yes |
+
+## Introduction
+
+When ypou cast an object you should be clear of your intentions. The regular cast will throw on `InvalidCastException` if the object is not of the type that ypu cast to. Using the `as` keyword does not throw this exception, but instead has the behavior that the resulting value can be null. This condition should be checked in your code, otherwise an `NullReferenceException` will be thrown.
+
+## How to solve
+
+When using the `as` keyword to cast on object, include a `null` check of the resulting variable.
+
+## Example
+
+Code that triggers a diagnostic:
+``` cs
+class BadExample {
+  public string BadMethod(object obj) {
+    return obj as string;
```

Comment:
I don't think this triggers it, because there's not a subsequent dereference.

Reply:
Correct, used str.Length now to trigger it and named the methods to make the point more clear